### ISSUE TITLE
Default content-type

### ISSUE THREAD

**Issue by chris-martin** [derekelkins/sws#20]
I've long lamented that I can't use sws to serve my blog, because my HTML files don't have an `.html` extension. On the public server, nginx is configured to give files without extensions a default content-type of `text/html`.

Thanks to #19, we can now do this:

```
sws --dev-mode --public -X"Content-Type: text/html; charset=utf-8"
```

But that isn't a default, it applies to *everything* -- so now my HTML works but my CSS does not.

Perhaps sws could have a `--default-content-type` option, or maybe even a `--default-html` option (if my use case is a common one?).

**Comment by derekelkins** [comment]
My first reaction is to add something like a `--content-type "htm=text/html"` that would assign content types based on extension. I guess it would make sense to add a `--default-content-type` to use as an alternative to whatever Warp falls back to when it doesn't know (presumably `application/octet-stream`) assuming there's an easy way to accomplish that.

**Comment by chris-martin** [comment]
Both of those would be good. I think I was unclear about what I'm really after: a rule that says "for files with *no* extension, set the content-type to `text/html; **charset=utf-8`"

**Comment by derekelkins** [comment]
The most natural solution would be to provide my own `getMimeType` in http://hackage.haskell.org/package/wai-middleware-static-0.8.2/docs/src/Network.Wai.Middleware.Static.html#local-6989586621679053266 in [`wai-middleware-static`](http://hackage.haskell.org/package/wai-middleware-static). Unfortunately, it is currently hard-coded, so I may make a pull request against `wai-middleware-static` to make it a parameter.

I can try passing in extra headers and hope it does the right thing (i.e. overrides earlier headers), but I suspect it will not.

**Comment by derekelkins** [comment]
I've made [a pull request](https://github.com/scotty-web/wai-middleware-static/pull/19) against `wai-middleware-static` to allow overriding the default `getMimeType` function. I've made changes locally to provide command-line options to override content types using the syntax above, e.g. `--content-type "htm=text/html"`. Your use case would be accomplished via `--content-type "=text/html"`. I'll push them and a new cabal package to Hackage once the changes to `wai-middleware-static` are accepted (or some analogue provided).
